docs: add raw policies documentation

fabriziosestito opened the pull request:
## Description

Adds raw policies documentation

TODO:
- [x] go tutorial: (see: https://github.com/kubewarden/raw-mutation-policy)
- [x] rust tutorial: (see: https://github.com/kubewarden/raw-validation-policy as reference)
- [x] WASI tutorial: (see: https://github.com/kubewarden/raw-validation-wasi-policy and https://github.com/kubewarden/raw-mutation-wasi-policy)
- [x] rego/opa tutorial (see https://github.com/kubewarden/raw-validation-opa-policy, OPA cannot mutate)

viccuad commented on docs/writing-policies/raw-policies.md:
```diff
@@ -0,0 +1,174 @@
+The Policy Server exposes the `/validate_raw` endpoint that can be used to validate
+arbitrary JSON documents against a Kubewarden policy.
+
+## Running the policy server outside Kubernetes
```
I think this page should be renamed "run policy-server outside of Kubernetes", and be kept in this "tutorial" section. But I don't think it should include additional info on raw policies.

viccuad commented on docs/writing-policies/raw-policies.md:
```diff
@@ -0,0 +1,174 @@
+
+Please refer to the [metadata](metadata.md) section for more information about policy metadata.
+
+## Using the raw validation endpoint
```
I think this section should not be here, but directly in the "specification" section.

viccuad commented on docs/writing-policies/raw-policies.md:
```diff
@@ -0,0 +1,174 @@
+
+- [Go](../writing-policies/go/10-writing-raw-policies.md)
+
+## Metadata
```
I think this section should not be here but in specification section, under "reference documentation -> policy specification", as this doesn't belong in a tutorial.

fabriziosestito commented:
@viccuad I've moved the raw policies guide from tutorials to howtos.
The goal is to give the user a howto on how _to execute_ raw policies, so I kept the "running policy server" related information there.

I've added spec/metadata info to the corresponding specification sections and changed specs with real-world commands in the how-to guide.

flavio reviewed (changes requested):
Great job, I love how you moved the contents since the original PR.

I left some minor suggestions

fabriziosestito commented on docs/howtos/raw-policies.md:
```diff
@@ -0,0 +1,274 @@
+
+## Running a Policy Server inside Kubernetes without the Kubewarden controller
+
+Alternatively, the Policy Server can be deployed in Kubernetes without the Kubewarden controller.
```
Nothing prevents the user from using the `validate_raw` endpoint of a controller-managed Policy server. The goal of this part of the how-to is to document alternative installation methods since the user might or might not want to install the full Kubewarden stack via the helm charts.

flavio replied:
No this is not possible, the problem is subtle. To host a raw policy the user must change the `policies.yml` file loaded by the Policy Server instance. However, this "file" is managed by our controller. Hence any change done to this file will then be reverted by our controller.

I wonder if we should ad that to the suggested text :thinking:

fabriziosestito replied:
Ah yes i totally missed that! I will apply the suggestion thanks!

fabriziosestito commented on docs/howtos/raw-policies.md:
````diff
@@ -0,0 +1,274 @@
+  }
+}'
+```
+
````
Yes, I feel that is a bit too much

flavio replied:
Fine, let's drop the suggestion :+1:

flavio reviewed (approved):
Looks great! Thanks for having included the feedback

viccuad commented on docs/howtos/raw-policies.md:
```diff
@@ -0,0 +1,283 @@
+    spec:
+      containers:
+        - name: policy-server
+          image: ghcr.io/kubewarden/policy-server:1.9.0
```
```suggestion
          image: ghcr.io/kubewarden/policy-server:v1.9.0
```

viccuad commented on docs/howtos/raw-policies.md:
```diff
@@ -0,0 +1,283 @@
+
+The raw validation endpoint is exposed at `/validate_raw` and accepts `POST` requests.
+
+Let's try to validate a JSON document against the `raw-validation` policy:
```
maybe worth it to provide people with a port-forward:
`kubectl port-forward service/policy-server-service 3000:3000 -n default`
